File: agent/library/src/main/java/org/matilda/services/plugins/InMemoryClassLoader.java
package org.matilda.services.plugins;

import java.util.Map;

public class InMemoryClassLoader extends ClassLoader {
    private final Map<String, byte[]> mClasses;

    public InMemoryClassLoader(Map<String, byte[]> classes) {
        mClasses = classes;
    }

    @Override
    protected Class<?> findClass(String name) throws ClassNotFoundException {
        if (mClasses.containsKey(name)) {
            byte[] classData = mClasses.get(name);
            return defineClass(name, classData, 0, classData.length);
        } else {
            throw new ClassNotFoundException(name);
        }
    }
}

File: agent/library/src/main/java/org/matilda/services/plugins/InMemoryClassLoader.kt
package org.matilda.services.plugins

class InMemoryClassLoader(private val mClasses: Map<String, ByteArray>) : ClassLoader() {
    override fun findClass(name: String): Class<*> {
        val classData = mClasses.getOrElse(name) {
            throw ClassNotFoundException(name)
        }
        return defineClass(name, classData, 0, classData.size)
    }
}
